ocaml-lsp-server/src/document.ml

```diff
@@ -308,7 +317,8 @@ module Merlin = struct
         | [] | (_, `Index _, _) :: _ -> None
         | (loc, `String typ, _) :: _ ->
           let doc = doc_comment pipeline pos in
-          Some { loc; typ; doc })
+          let syntax_doc_result = syntax_doc pipeline pos in
```

Comment:
This means that Merlin will have to perform extra work that most of the time gets thrown away. We should not call `syntax_doc` if the option is not enabled and always return `false` in that case.

Reply:
Thanks for pointing this out